packages/flutter_tools/lib/src/isolated/devtools_launcher.dart

```diff
@@ -15,47 +18,75 @@ import '../resident_runner.dart';
 /// a devtools dep in google3.
 class DevtoolsServerLauncher extends DevtoolsLauncher {
   DevtoolsServerLauncher({
+    @required ProcessManager processManager,
+    @required String pubExecutable,
     @required Logger logger,
-  }) : _logger = logger;
+  })  : _processManager = processManager,
+        _pubExecutable = pubExecutable,
+        _logger = logger;
 
+  final ProcessManager _processManager;
+  final String _pubExecutable;
   final Logger _logger;
 
-  io.HttpServer _devtoolsServer;
+  io.Process _devToolsProcess;
+  Uri _devToolsUri;
+
+  static final RegExp _devToolsPattern =
+      RegExp(r'Serving DevTools at ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
 
   @override
-  Future<void> launch(Uri observatoryAddress) async {
+  Future<void> launch(Uri vmServiceUri) async {
     try {
-      await serve();
-      await devtools_server.launchDevTools(
-        <String, dynamic>{
-          'reuseWindows': true,
-        },
-        observatoryAddress,
-        'http://${_devtoolsServer.address.host}:${_devtoolsServer.port}',
-        false,  // headless mode,
-        false,  // machine mode
-      );
+      _devToolsProcess = await _processManager.start(<String>[
```

Comment:
I think this needs a `pub global activate` too, otherwise it could fail right?

Reply:
added call to `pub global activate devtools`, and called `pub global run devtools` in the onDone callback. PTAL.